src/support_code_library_builder/types.ts

```diff
@@ -22,10 +22,10 @@ export interface ITestStepHookParameter {
   testStepId: string
 }
 
-export type TestCaseHookFunctionWithoutParameter = () => void | Promise<void>
+export type TestCaseHookFunctionWithoutParameter = () => any | Promise<any>
 export type TestCaseHookFunctionWithParameter = (
   arg: ITestCaseHookParameter
-) => void | Promise<void>
+) => any | Promise<any>
```

Comment:
`any` is obviously a bit broad but I think it's faithful to the API, in the sense that you _can_ return/resolve anything you want from a hook or step function, but `skipped` and `pending` have special meaning.